src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,9 +33,65 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        
+        String[] addressComponents = trimmedAddress.split(", ");
+       
+        Block block = new Block(Integer.parseInt(addressComponents[BLOCK_INDEX]));
+        Street street = new Street(addressComponents[STREET_INDEX]);
+        Unit unit = new Unit(addressComponents[UNIT_INDEX]);
+        PostalCode postal = new PostalCode(Integer.parseInt(addressComponents[POSTAL_CODE_INDEX]));
+        
+        this.value = block.getBlockNo() + COMMA + street.getStreetName() + COMMA
+        			 + unit.getUnitNo() + COMMA + postal.getPostalCode();
+    }
+    
+    private class Block {
```

Comment:
When you keep the classes private, you disallow any other classes to speak to the classes, except for Address. So now, if there is a need to just retrieve the block number of the user, how can it be done? Note that Address does not expose any method to retrieve the individual values.